### ISSUE TITLE
RunSemaphoreSlimTest8_ConcWaitAsyncAndRelease failed in Windows_Debug OuterLoop

### ISSUE THREAD

**Issue by mellinoe** [dotnet/corefx#6410]
See this run here: http://dotnet-ci.cloudapp.net/job/dotnet_corefx/job/outerloop_win10_debug/lastCompletedBuild/testReport/System.Threading.Tests/SemaphoreSlimTests/RunSemaphoreSlimCurrentTests/

````
Stacktrace

MESSAGE:
Assert.Equal() Failure\r\nExpected: 50\r\nActual: 46
+++++++++++++++++++
STACK TRACE:
at System.Threading.Tests.SemaphoreSlimTests.RunSemaphoreSlimTest8_ConcWaitAsyncAndRelease(Int32 initial, Int32 maximum, Int32 waitThreads, Int32 releaseThreads, Int32 succeededWait, Int32 failedWait, Int32 finalCount, Int32 timeout) in d:\j\workspace\outerloop_win---0cba2915\src\System.Threading\tests\SemaphoreSlimTests.cs:line 586 at System.Threading.Tests.SemaphoreSlimTests.RunSemaphoreSlimCurrentTests() in d:\j\workspace\outerloop_win---0cba2915\src\System.Threading\tests\SemaphoreSlimTests.cs:line 173
````

**Comment by stephentoub** [comment]
Dup of #6399 which was addressed by #6400

**Comment by mellinoe** [comment]
My mistake, searched for the wrong method name. Thanks for already fixing this :smile:
